rmgpy/solver/base.pyx

```diff
@@ -583,14 +762,124 @@ cdef class ReactionSystem(DASx):
+                                maxSurfaceAccumReactionIndex = i
+                    maxSurfaceAccumReaction = surfaceReactions[maxSurfaceAccumReactionIndex]
+                elif zeroProduction:
+                    for i in xrange(len(surfaceReactionIndices)):
+                        index = surfaceReactionIndices[i]
+                        maxSurfaceReactionAccum = 0.0
+                        maxSurfaceAccumReactionIndex = -1
+                        if surfaceInfAccumNumIndex in productIndices[index]:
+                            reactionRate = coreReactionRates[index]
+                            if reactionRate > maxEdgeReactionAccum:
+                                maxSurfaceReactionAccum = reactionRate
+                                maxSurfaceAccumReactionIndex = i
+                    maxSurfaceAccumReaction = surfaceReactions[maxSurfaceAccumReactionIndex]
+                elif len(surfaceTotalDivAccumNums)>0: 
+                    maxSurfaceAccumReactionIndex = numpy.argmax(surfaceTotalDivAccumNums)
+                    maxSurfaceAccumReaction = surfaceReactions[maxSurfaceAccumReactionIndex]
+                    maxSurfaceDifLnAccumNum = numpy.log(surfaceTotalDivAccumNums[maxSurfaceAccumReactionIndex])
+            else:
+                maxSurfaceAccumReactionIndex = -1
+                maxSurfaceAccumReaction = None
+                maxSurfaceDifLnAccumNum = 0.0
+                
+            #manage surface reaction movement "on the fly"
+            if maxSurfaceAccumReaction and maxSurfaceDifLnAccumNum > toleranceMoveSurfaceReactionToCore:
+                logging.info('Moving reaction {0} from surface to core'.format(maxSurfaceAccumReaction))
+                surfaceReactions.remove(maxSurfaceAccumReaction)
+                surfaceSpecies,surfaceReactions =self.initialize_surface(coreSpecies,coreReactions,surfaceSpecies,surfaceReactions)
+                surfaceSpeciesIndices = self.surfaceSpeciesIndices
+                surfaceReactionIndices = self.surfaceReactionIndices
+                logging.info('Surface has {0} Species and {1} Reactions'.format(len(surfaceSpeciesIndices),len(surfaceReactionIndices)))
+            
+            #surface flux calculations
+            surfaceSpeciesRates = numpy.zeros(len(surfaceSpeciesIndices))
+            surfaceSpeciesProduction = coreSpeciesProductionRates[surfaceSpeciesIndices]
+            surfaceSpeciesConsumption = coreSpeciesConsumptionRates[surfaceSpeciesIndices]
+            for i in xrange(len(surfaceSpeciesIndices)):
+                surfaceSpeciesRates[i] = max(abs(surfaceSpeciesProduction[i]),abs(surfaceSpeciesConsumption[i]))
+                
+            if len(surfaceSpeciesIndices) > 0:
+                maxSurfaceSpeciesIndex = numpy.argmax(surfaceSpeciesRates)
+                maxSurfaceSpecies = surfaceSpecies[maxSurfaceSpeciesIndex]
+                maxSurfaceSpeciesRate = surfaceSpeciesRates[ind]
```

Comment:
`ind` is not defined anywhere. I just happened to notice the warning while compiling.